src/game/logic/object/locomotor.cpp

```diff
@@ -298,7 +298,7 @@ void LocomotorStore::Reset()
     auto it = m_locomotorTemplates.begin();
 
     for (;;) {
-        if (!(it != m_locomotorTemplates.end())) {
+        if (it == m_locomotorTemplates.end()) {
```

Comment:
all of this could be simplified with std::remove_if:
```c++
m_locomotorTemplates.erase(std::remove_if(m_locomotorTemplates.begin(), m_locomotorTemplates.end(), [](auto template) { return t.second->DeleteOverrides() == nullptr; });
```
Also shouldn't the iterator be assigned to the result of `.erase` here?

Reply:
On that note, this algorithm looks suspicious.

Down below I would have expected
```c++
it = m_locomotorTemplates.erase(it);
```

Essentially loop will continue and then reference invalidated iterator. Looks like bug.